runtime/lua/vim/lsp/log.lua

```diff
@@ -1,41 +1,65 @@
 -- Logger for language client plugin.
 
-local log = {}
+local loggers = {}
 
--- Log level dictionary with reverse lookup as well.
---
--- Can be used to lookup the number from the name or the name from the number.
--- Levels by name: 'trace', 'debug', 'info', 'warn', 'error'
--- Level numbers begin with 'trace' at 0
-log.levels = {
+local raw_levels = {
   TRACE = 0;
   DEBUG = 1;
   INFO  = 2;
   WARN  = 3;
   ERROR = 4;
 }
 
--- Default log level is warn.
-local current_log_level = log.levels.WARN
+-- Log level dictionary with reverse lookup as well.
+--
+-- Can be used to lookup the number from the name or the name from the number.
+-- Levels by name: 'trace', 'debug', 'info', 'warn', 'error'
+-- Level numbers begin with 'trace' at 0
+local levels = vim.deepcopy(raw_levels)
+vim.tbl_add_reverse_lookup(levels)
+
 local log_date_format = "%FT%H:%M:%SZ%z"
 
-do
+
+local function create_logger(filename)
```

Comment:
Should I rename this to `new_logger` to be more in line with other function signatures?